### PR TITLE
fix: reject empty explicit subscription filters

### LINKED ISSUE #46: Reject empty explicit subscription filter inputs
## Summary
When callers explicitly choose `--filter-json`, `--filter-file`, or `--filter-stdin`, empty content should be rejected instead of being normalized to `{}`.

## Current behavior
`parseJsonArg()` treats falsy input as an empty object. After PR #45 this means these cases can silently create a broad unfiltered subscription:

- `agentinbox subscription add <sourceId> --filter-stdin` with empty stdin or a failed upstream pipe
- `agentinbox subscription add <sourceId> --filter-file empty.json`
- potentially empty `--filter-json` payload handling depending on shell invocation

## Expected behavior
- no filter flag at all: still means `{}`
- explicit filter input mode selected: require a non-empty JSON object payload
- return a clear client error that identifies the failing input surface

## Why it matters
The feature in PR #45 is meant to make filter input more robust. Silent fallback to `{}` turns quoting or pipeline mistakes into catch-all subscriptions, which is the opposite failure mode.

## Follow-up
- update CLI parsing for explicit filter modes
- add regression tests for empty file/stdin cases
- update docs if needed

### PR DESCRIPTION
**Pull request by jolestar**
## Summary
- reject empty explicit filter payloads for `--filter-file` and `--filter-stdin` instead of silently normalizing them to `{}`
- keep the no-filter default unchanged, so omitting filter flags still means an empty filter object
- add CLI regression coverage and document the explicit-input requirement

Closes #46

## Testing
- npm run build
- npm test

### CODE REVIEW

**Comment by jolestar** [comment]
I think there is still one substantive issue here before merge.

The PR goal is to reject accidentally broad subscriptions when the caller explicitly selects `--filter-file` or `--filter-stdin` but provides an empty filter. Right now the implementation only rejects missing/blank payloads, but it still accepts an explicit empty object (`{}`).

That leaves the same practical footgun in place for one common case:
- `echo "{}" > filter.json` + `--filter-file filter.json`
- `printf "{}" | agentinbox subscription add ... --filter-stdin`

Both still create an unfiltered subscription, even though the docs and error text now say `non-empty JSON object`.

So I think this should do one of two things consistently:
1. Actually reject `{}` as well when `--filter-file` / `--filter-stdin` are used, and add regression coverage for that case.
2. Or, if `{}` is intentionally allowed, then the docs/error wording should stop claiming `non-empty JSON object`.

Given this PR is specifically fixing issue #46, I think option 1 is the right behavior.

File: src/util.ts [hunk 1]
```diff
@@ -8,10 +8,22 @@ export function generateId(prefix: string): string {
...
+): Record<string, unknown> {
   if (!raw) {
+    if (options?.requireNonEmpty) {
+      throw new Error(`invalid ${source}: expected a non-empty JSON object`);
+    }
     return {};
   }
+  if (options?.requireNonEmpty && raw.trim() === "") {
+    throw new Error(`invalid ${source}: expected a non-empty JSON object`);
+  }
```
[the remark this answers is not in the document]
Comment (jolestar) [reply src/util.ts:26]: Fixed in f6d0b27. Explicit filter modes now enforce a non-empty JSON object, so `{}` is rejected alongside blank input when `--filter-json`, `--filter-file`, or `--filter-stdin` is used.

File: test/control_plane.test.ts [hunk 2]
```diff
@@ -731,6 +731,27 @@ test("cli subscription add accepts filter-file and filter-stdin and echoes norma
...
+    assert.match(emptyFile.stderr, /invalid filter file .*empty-filter\.json: expected a non-empty JSON object/);
+
+    const emptyStdin = runCli([
+      "subscription",
+      "add",
+      source.sourceId,
+      "--filter-stdin",
+    ], env, 25_000, "");
+    assert.notEqual(emptyStdin.status, 0);
+    assert.match(emptyStdin.stderr, /invalid stdin filter: expected a non-empty JSON object/);
```
[the remark this answers is not in the document]
Comment (jolestar) [reply test/control_plane.test.ts:776]: Fixed in f6d0b27. I added regression assertions for explicit empty-object payloads as well: `--filter-json {}`, a file containing `{}`, and stdin containing `{}` all now fail with the same non-empty-object error.

File: docs/site/reference/cli.md [hunk 3]
```diff
@@ -71,7 +71,8 @@ agentinbox subscription reset <subscription_id> --start-policy latest
 
 `subscription add` accepts exactly one of `--filter-json`, `--filter-file`, or
 `--filter-stdin`. The created subscription response echoes the normalized
-persisted `filter`.
+persisted `filter`. When `--filter-file` or `--filter-stdin` is selected, the
+supplied payload must be a non-empty JSON object.
```
[the remark this answers is not in the document]
Comment (jolestar) [reply docs/site/reference/cli.md]: Fixed in f6d0b27. The docs now match the enforced behavior: any explicit filter input mode (`--filter-json`, `--filter-file`, or `--filter-stdin`) requires a non-empty JSON object.
